xyz-hub-service/src/main/java/com/here/xyz/httpconnector/config/JDBCImporter.java

```diff
@@ -0,0 +1,396 @@
+                + " aws_commons.create_s3_uri( \n"
+                + "     '{s3Bucket}', \n"
+                + "     '{s3Path}', \n"
+                + "     '{s3Region}' \n"
+                + " )),'{s3Path}:{partSize}' as iml_import_hint")
+                .replace("{partSize}",Long.toString(curFileSize))
+                .replace("{s3Bucket}",s3Bucket)
+                .replace("{s3Path}",s3Path)
+                .replace("{s3Region}",s3Region)
+                .replace("{columns}", csvFormat.equals(CSVFormat.GEOJSON) ? "jsondata" : "jsondata,geo"));
+
+        q.setVariable("columns", tablename);
+        q.setVariable("table", tablename);
+        q.setVariable("schema", schema);
+        q.substitute();
+
+        logger.info("Execute S3-Import {}->{} {}", tablename, s3Path, q.text());
+        return getClient(clientID).query(q.text())
+                .execute()
+                .map(row -> row.iterator().next().getString(0));
+    }
+
+    /**
+     * Finalize Import - create Indices and drop temporary resources
+     */
+    public static Future<Void> finalizeImport(Job job, String schema){
+        String clientId = job.getTargetConnector();
+        String tableName = job.getTargetTable();
+
+        Promise<Void> p = Promise.promise();
+
+        List<Future> indicesFutures = new ArrayList<>();
+
+        for (String idxName: DEFAULT_IDX_LIST) {
+            SQLQuery q = createIdxQuery(idxName, schema, tableName);
+            indicesFutures.add(createIndex(clientId, q, idxName)
+                    .onSuccess(idx -> {
+                        logger.info("IDX creation of '{}' succeeded!", idx);
+                        ((Import)job).addIdx(idxName);
+                        //TODO: Add Manual Indices!
+                    }).onFailure(e -> {
+                        logger.warn("IDX creation '{}' failed! ",idxName,e);
+                        job.setErrorDescription(Import.ERROR_DESCRIPTION_IDX_CREATION_FAILED);
+                    }));
+        }
+
+        CompositeFuture.join(indicesFutures).onComplete(
+                t-> {
+                    deleteImportTrigger(clientId, schema, tableName);
+                    p.complete();
+                }
+        ).onFailure(e -> {
+            logger.warn("Table cleanup has failed",e);
+            if(job.getErrorDescription() == null)
+                job.setErrorDescription(Import.ERROR_DESCRIPTION_TABLE_CLEANUP_FAILED);
+        });
+
+        return p.future();
+    }
+
+    public static Future<Void> deleteView(String clientID, String schema, String tableName){
+        SQLQuery q = new SQLQuery("DROP VIEW IF EXISTS ${schema}.${viewname};");
+
+        q.setVariable("schema",schema);
+        q.setVariable("viewname", getViewName(tableName));
+        q.substitute();
+
+        logger.info("Delete view {}",getViewName(tableName));
+
+        return getClient(clientID).query(q.text())
```

Comment:
Also #substitute() is chainable. It could be used right before text() in general, because it's basically always the last step before executing the query.

`e.g. query(q.substitute().text)`